tests/unit/util/pack_traversal.cpp

```diff
@@ -0,0 +1,621 @@
+//  Copyright (c) 2017 Denis Blank
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithm>
```

Comment:
In my experience, for MSVC, including the HPX headers first avoids various warnings.